README.md

```diff
@@ -300,6 +300,294 @@ schemas:
+
+Example:
+```yml
+schemas:
+  bookings:
+    array_of: booking
+    date: booking_date
+    document: id
+    products:
+      array_of:
+      accomodation:
+        path: ../hotel[booking_id=<link>]/accomodation
+        link: id
+```
+```xml
+<xml>
+  <booking>
+    <id>1</id>
+    <booking_date>2020-01-01</booking_date>
+  </booking>
+  <booking>
+    <id>2</id>
+    <booking_date>2020-01-02</booking_date>
+  </booking>
+  <hotel>
+    <booking_id>1</booking_id>
+    <accomodation>Standard</accomodation>
+  </hotel>
+  <hotel>
+    <booking_id>2</booking_id>
+    <accomodation>Premium</accomodation>
+  </hotel>
+</xml>
+```
+```ruby
+{
+  bookings: [
+    {
+      date: "2020-01-01",
+      document: "1"
+      products: [
+        { accomodation: "Standard" }
+      ]
+    },
+    {
+      date: "2020-01-02",
+      document: "2"
+      products: [
+        { accomodation: "Premium" }
+      ]
+    }
+  ]
+}
+```
+
+In this example if I didn't use the `link` to get only the hotel of each booking, it would have returned two accomodations for each booking and instead of extract a string with the accomodation it would extract an array with all the accomodations for each booking.
+
+You can combine the `link` with `array_of` if you want search for a list of elements filtering by some field, just provide the `path` and the `link`:
+
+```yml
+schemas:
+  bookings:
+    array_of: booking
+    date: date
+    document: id
+    products:
+      array_of:
+        path: ../products[booking_id=<link>]
+        link: id
+      ....
+```
+
+### uniq_by
+
+Can only be used with **array_of**.
+
+Has a similar behavior like Ruby **uniq** method on arrays, do you provide a path to the `uniq_by` tag and then we will keep only on the array_of the paths that have a diferrent values in the path provided on the tag `uniq_by`, if the value of the path provided in the tag `uniq_by` is equal in more than one path, we will keep the first path and the remaining paths will be ignored. This functionality is useful when some node in the XML is duplicated and it is only necessary to obtain the info from one of the nodes.
+
+```yml
+schemas:
+  bookings:
+    array_of:
+      path: booking
+      uniq_by: id
+    date: bdate
+    document: id
+```
+```xml
+<xml>
+  <booking>
+    <id>1</id>
+    <bdate>2020-01-01</bdate>
+  </booking>
+  <booking>
+    <id>1</id>
+    <bdate>2020-01-01</bdate>
+  </booking>
+</xml>
+```
+```ruby
+{
+  bookings: [
+    {
+      date: "2020-01-01",
+      document: "1"
+    }
+  ]
+}
+```
+
+In this example if we don't use the tag `uniq_by` there would be extracted two elements with the same data, like:
+
+```ruby
+{
+  bookings: [
+    {
+      date: "2020-01-01",
+      document: "1"
+    },
+    {
+      date: "2020-01-01",
+      document: "1"
+    }
+  ]
+}
+```
+
+### array_presence: first_only
```

Comment:
for future improvements, I believe this could be changed to `first_only: true`

Reply:
Yes, it's on my radar. As soon as I start working on #2 I will make the change